Dockerfile

```diff
@@ -1,16 +1,18 @@
-FROM golang:1.20-alpine3.18 as builder
+FROM --platform=$BUILDPLATFORM golang:1.20-alpine3.18 as builder
 WORKDIR $GOPATH/src/go.k6.io/k6
-COPY . .
+ARG TARGETOS TARGETARCH
```

Comment:
Not sure if this is a big issue, but just stressing that this change makes the Dockerfile unbuildable without providing these args and think like ` docker build . -t k6-test` won't work anymore :cry:

Reply:
>docker build . -t k6-test

Do we need to support it? I guess these PRs are setting the requirement for `buildx`. Do you see issues in doing it?

Reply:
It was just nice to have the ability to build an image for local testing without digging into what you should pass.  On the other side, it's a rare case. 

However, maybe a comment with the command or better fixing the Makefile's target (which also) resolves my concern.

Reply:
This actually works for me ... with `podman`. 

My problem was https://github.com/grafana/k6/pull/3323/files#r1321478143 

What error did you get ?

Reply:
I get the same error as @olegbespalov 

```
docker build --rm -t testbuildclassic  .
Sending build context to Docker daemon  93.62MB
Step 1/23 : FROM --platform=$BUILDPLATFORM golang:1.20-alpine3.18 as builder
failed to parse platform : "" is an invalid component of "": platform specifier component must match "^[A-Za-z0-9_-]+$": invalid argument
```

Reply:
From the google searches I did ... it seems like this should work for you too :thinking: although maybe `moby` and what `docker` is for most people is now very different :shrug: . 

Does this work if you run `DOCKER_BUILDKIT=1 docker build ...` ?

Reply:
I guess we are getting the error because we have a version installed of Docker Engine `< 23.0`. @olegbespalov do you confirm that it is the case for you as well?

Reply:
That's correct :+1: